src/scripts/main.js

```diff
@@ -1,3 +1,290 @@
 'use strict';
 
-// write your code here
+const cellsArray = [...document.querySelectorAll('.field-cell')];
+const usedCells = [];
+const startButton = document.querySelector('button.button');
+const startMessage = document.querySelector('.message-start');
+const loseMessage = document.querySelector('.message-lose');
+const winMessage = document.querySelector('.message-win');
+const score = document.querySelector('.game-score');
+const transitionTime = 300;
+let isWinMessage = false;
+
+function randomizer(numberToRandom = 9) {
+  return Math.floor(Math.random() * numberToRandom);
+}
+
+function changeStartButton(newClass) {
+  const startButtonClasses = startButton.classList;
+
+  startButtonClasses.remove(startButtonClasses[startButtonClasses.length - 1]);
+  startButtonClasses.add(newClass.toLowerCase());
+  startButton.innerText = newClass;
+}
+
+function addNumberToFreeCell(cellNumber) {
+  for (let i = 0; i < cellNumber; i++) {
+    const runningCell = document.createElement('div');
+    const freeCells = cellsArray.filter(cell => !usedCells.includes(cell));
+    const randomCell = freeCells[randomizer(freeCells.length - 1)];
+    const randomCellClass = randomizer() === 4
+      ? 'running-cell--4' : 'running-cell--2';
+
+    usedCells.push(randomCell);
+    runningCell.classList.add('running-cell', randomCellClass);
+    runningCell.style.top = `${randomCell.offsetTop}px`;
+    runningCell.style.left = `${randomCell.offsetLeft}px`;
+    runningCell.innerText = randomCellClass.slice(14);
+    cellsArray[0].parentElement.append(runningCell);
+  }
+};
+
+function clearGame() {
+  [...document.querySelectorAll('.running-cell')].map(cell => cell.remove());
+  isWinMessage = false;
+  winMessage.classList.add('hidden');
+  loseMessage.classList.add('hidden');
+  usedCells.length = 0;
+  score.innerHTML = 0;
+};
+
+function checkForLose() {
+  const tops = [10, 95, 180, 265];
+  const runningCellsArray = [...document.querySelectorAll('.running-cell')];
+
+  if (runningCellsArray.length === 16) {
+    const filteredAndSortedRunCells = [];
+
+    tops.forEach(topValue => runningCellsArray
+      .filter(cell => cell.offsetTop === topValue)
+      .sort((a, b) => a.offsetLeft - b.offsetLeft)
+      .forEach(cell => filteredAndSortedRunCells.push(cell))
+    );
+
+    let canContinueGame = false;
+
+    for (let i = 0; i < filteredAndSortedRunCells.length; i++) {
+      const cell = filteredAndSortedRunCells[i];
+      const nextCell = filteredAndSortedRunCells[i + 1];
+      const nextUpCell = filteredAndSortedRunCells[i + 4];
+
+      if (nextCell !== undefined && nextCell.offsetTop === cell.offsetTop
+        && nextCell.innerText === cell.innerText) {
```

Comment:
Long expression move to variable with meaningful variable naming. Fix all cases
```suggestion
      if (nextCell && nextCell.offsetTop === cell.offsetTop
        && nextCell.innerText === cell.innerText) {
```

Reply:
done